decent-dao/fractal-interface
# Gnosis module types

## Description
This PR adds in functionality for the webapp to detect the type of modules that are attached to a Gnosis Safe DAO.

## Notes
 This is done by querying events from the GnosisModuleProxyFactory. The module that is being inspected is checked to see if an event was emitted from that factory, if so, get the implementation contract address, and see if that matches any of our implementation contract addresses. If it does, we know the module is of that type. If not, we assign the "unknown" module type.


## Issue / Notion doc (if applicable)
https://github.com/decent-dao/fractal-interface/issues/591


## Testing
1) Create / browse to a Gnosis Safe DAO that has an attached Usul module.
2) Verify that on the DAO homepage, the "module types" section shows the correct contract address and module type


## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/84364476/198634722-af306601-34e9-4b80-8c90-a68c29cf3d69.png)
